core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2186,13 +2186,22 @@ class SparkContext(config: SparkConf) extends Logging {
    * has overridden the call site using `setCallSite()`, this will return the user's version.
    */
   private[spark] def getCallSite(): CallSite = {
-    lazy val callSite = Utils.getCallSite()
-    CallSite(
-      Option(getLocalProperty(CallSite.SHORT_FORM)).getOrElse(callSite.shortForm),
-      Option(getLocalProperty(CallSite.LONG_FORM)).getOrElse(callSite.longForm)
-    )
+    if (getLocalProperty(CallSite.SHORT_FORM) == null
```

Comment:
You can probably simplify by creating two Options for the two properties as vals first